File: Frontend/Shawarma-O’Clock-frontend/shawarma-frontend/src/services/orderService.js
// src/services/orderService.js
import apiClient from './apiClient';

const orderService = {
  // Place new order
  async placeOrder(userId, totalAmount, orderDetails) {
    try {
      const response = await apiClient.post('/Order', {
        userID: userId,
        totalAmount: totalAmount,
        orderDetails: orderDetails.map(item => ({
          itemID: item.itemID,
          quantity: item.quantity,
          subtotal: item.subtotal,
        })),
      });
      return response.data;
    } catch (error) {
      throw error.response?.data?.message || 'Failed to place order';
    }
  },

  // Get all orders with customer details (Admin only) - ENHANCED
  async getAllOrders() {
    try {
      const response = await apiClient.get('/Order');
      return response.data;
    } catch (error) {
      throw error.response?.data?.message || 'Failed to fetch orders';
    }
  },

  // Get orders by user
  async getOrdersByUser(userId) {
    try {
      const response = await apiClient.get(`/Order/user/${userId}`);
      return response.data;
    } catch (error) {
      throw error.response?.data?.message || 'Failed to fetch user orders';
    }
  },

  // Get order details with items - ENHANCED
  async getOrderDetails(orderId) {
    try {
      const response = await apiClient.get(`/Order/${orderId}/details`);
      return response.data;
    } catch (error) {
      throw error.response?.data?.message || 'Failed to fetch order details';
    }
  },

  // Update order status (Admin only)
  async updateOrderStatus(orderId, status) {
    try {
      const response = await apiClient.patch(`/Order/${orderId}/status?status=${status}`);
      return response.data;
    } catch (error) {
      throw error.response?.data?.message || 'Failed to update order status';
    }
  },

  // Search orders (Admin only) - NEW
  async searchOrders(searchTerm = null, status = null, fromDate = null, toDate = null) {
    try {
      const params = new URLSearchParams();
      if (searchTerm) params.append('searchTerm', searchTerm);
      if (status && status !== 'All') params.append('status', status);
      if (fromDate) params.append('fromDate', fromDate);
      if (toDate) params.append('toDate', toDate);
      
      const queryString = params.toString() ? `?${params.toString()}` : '';
      const response = await apiClient.get(`/Order/search${queryString}`);
      return response.data;
    } catch (error) {
      throw error.response?.data?.message || 'Failed to search orders';
    }
  },

  // Get order statistics (Admin only) - NEW
  async getOrderStatistics() {
    try {
      const response = await apiClient.get('/Order/statistics');
      return response.data;
    } catch (error) {
      throw error.response?.data?.message || 'Failed to fetch order statistics';
    }
  },
};

export default orderService;

File: Backend/ShawarmaBackend/Services/Implementations/OrderService.cs
using ShawarmaBackend.Models.Domain;
using ShawarmaBackend.Models.DTOs.Requests;
using ShawarmaBackend.Repositories.Interfaces;
using ShawarmaBackend.Services.Interfaces;

namespace ShawarmaBackend.Services.Implementations
{
    public class OrderService : IOrderService
    {
        private readonly IOrderRepository _orderRepository;
        private readonly IMenuRepository _menuRepository;

        public OrderService(IOrderRepository orderRepository, IMenuRepository menuRepository)
        {
            _orderRepository = orderRepository;
            _menuRepository = menuRepository;
        }

        public async Task<(int orderId, string message)> PlaceOrder(PlaceOrderRequest request)
        {
            // Validate order has items
            if (request.OrderDetails == null || !request.OrderDetails.Any())
            {
                return (-1, "Order must contain at least one item");
            }

            // Validate stock availability
            foreach (var item in request.OrderDetails)
            {
                var menuItem = await _menuRepository.GetMenuItemById(item.ItemID);

                if (menuItem == null)
                {
                    return (-1, $"Item with ID {item.ItemID} not found");
                }

                if (!menuItem.IsAvailable)
                {
                    return (-1, $"{menuItem.Name} is not available");
                }

                if (menuItem.Quantity < item.Quantity)
                {
                    return (-1, $"Insufficient stock for {menuItem.Name}. Available: {menuItem.Quantity}");
                }
            }

            // Calculate and validate total amount
            decimal calculatedTotal = request.OrderDetails.Sum(x => x.Subtotal);
            if (Math.Abs(calculatedTotal - request.TotalAmount) > 0.01m)
            {
                return (-1, "Total amount mismatch");
            }

            // Place order
            return await _orderRepository.PlaceOrder(request);
        }

        public async Task<List<Order>> GetAllOrders()
        {
            return await _orderRepository.GetAllOrders();
        }

        public async Task<List<Order>> GetOrdersByUser(int userId)
        {
            return await _orderRepository.GetOrdersByUser(userId);
        }

        public async Task<List<OrderDetail>> GetOrderDetails(int orderId)
        {
            return await _orderRepository.GetOrderDetails(orderId);
        }

        public async Task<(int orderId, string status, string message)> UpdateOrderStatus(int orderId, string status)
        {
            // Validate status
            var validStatuses = new[] { "Pending", "Preparing", "Ready", "Completed", "Cancelled" };
            if (!validStatuses.Contains(status))
            {
                return (-1, status, "Invalid status");
            }

            return await _orderRepository.UpdateOrderStatus(orderId, status);
        }
    }
}